### PR TITLE
[Enhencement](csv_reader) support quotes content include delimiter and escape character

### PR DESCRIPTION
**Pull request by ZhangYu0123**
Support quotes content include delimiter and escape character.
split line like this:
     "9","10","3","ab,c" 

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.9 seconds
 stream load tsv:          497 seconds loaded 74807831229 Bytes, about 143 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230302165603_clickbench_pr_107572.html

**Comment by morningman** [comment]
Hi @ZhangYu0123 , I saw you submit several PR with same issue, and the others are closed.
What is difference among this PR?

**Comment by ZhangYu0123** [comment]
> Hi @ZhangYu0123 , I saw you submit several PR with same issue, and the others are closed. What is difference among this PR?

The code are the same, but
the other PR’s regression tests are not passed cased by the nereidsp0 test.

**Comment by morningman** [comment]
> > Hi @ZhangYu0123 , I saw you submit several PR with same issue, and the others are closed. What is difference among this PR?
> 
> The code are the same, but the other PR’s regression tests are not passed cased by the nereidsp0 test.

You don't need to submit a new PR, you can just update your branch, the PR will be updated automatically

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 1]
```diff
@@ -462,8 +462,22 @@ void CsvReader::_split_line_for_single_char_delimiter(const Slice& line) {
...
+            if (_trim_double_quotes && *(value + cur_pos) == '\"') {
+                if (cur_pos == start_field) {
+                    // start_field is ", mark quoted true
+                    quoted = true;
+                } else if (*(value + cur_pos - 1) != '\\' && quoted) {
+                    // cur_pos is ", mark quoted finished
+                    quoted = !quoted;
+                }
+                continue;
+            } else if (_trim_double_quotes && quoted) {
```
Comment (morningman) [inline be/src/vec/exec/format/csv/csv_reader.cpp:477]: Better leave `_split_line_for_single_char_delimiter` fast enough to not handle this complex situation.
Maybe we should be a new parameter to specify the `escape` explicitly.

**Comment by ZhangYu0123** [comment]
run buildall

File: be/src/vec/exec/format/csv/csv_reader.cpp (hunk 1 above)
Comment (ZhangYu0123) [reply be/src/vec/exec/format/csv/csv_reader.cpp]: > 



> Better leave `_split_line_for_single_char_delimiter` fast enough to not handle this complex situation. Maybe we should be a new parameter to specify the `escape` explicitly.

done

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run buildall

**Pull request closed without merging**